store/store_test.go

```diff
@@ -243,3 +243,38 @@ func TestMetadata(t *testing.T) {
 	require.Error(err)
 	require.Nil(v)
 }
+
+func TestExtendedCommits(t *testing.T) {
+	t.Parallel()
+
+	require := require.New(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	kv, err := NewDefaultInMemoryKVStore()
+	require.NoError(err)
+	s := New(kv)
+
+	// reading before saving returns error
+	commit, err := s.GetExtendedCommit(ctx, 1)
+	require.Error(err)
+	require.ErrorIs(err, ds.ErrNotFound)
+	require.Nil(commit)
+
+	expected := &abcitypes.ExtendedCommitInfo{
+		Round: 123,
+		Votes: []abcitypes.ExtendedVoteInfo{{
+			Validator: abcitypes.Validator{
+				Address: types.GetRandomBytes(20),
+				Power:   123,
+			},
+			VoteExtension:      []byte("extended"),
+			ExtensionSignature: []byte("just trust me bro"),
+			BlockIdFlag:        0,
+		}},
+	}
+
+	err = s.SaveExtendedCommit(ctx, 10, expected)
+	require.NoError(err)
```

Comment:
i would add `s.GetExtendedCommit(ctx, 10)` to make sure that you can Get the saved extended commit here.